recipe/magento2.php

```diff
@@ -229,6 +229,7 @@
 set('artifact_file', 'artifact.tar.gz');
 set('artifact_dir', 'artifacts');
 set('artifact_excludes_file', 'artifacts/excludes');
+set('build_from_repo', false);
```

Comment:
A comment should be added here so that an explanation is included in the generated documentation. Also,  `set('repository', null)` with another comment. Otherwise this is going to be a secret hidden feature

Reply:
> A comment should be added here so that an explanation is included in the generated documentation. Also, `set('repository', null)` with another comment. Otherwise this is going to be a secret hidden feature

Seems right, I'll add these and push the modifications.

About the variable name (`build_from_repo`), does it seem right to you?
English is not my native language so I'm not good at naming things :sweat_smile: 

>  I'm not totally sure I understand what happens here

The current artifact mechanism only allows to build the artifact from the current working directory.
It works fine on CIs, because when a pipeline is executed, the repository is automatically cloned, and the artifact is built from there.

But if you need to deploy from a local environment (which runs in developer mode and contains lots of things that definitely shouldn't be bundled in the artifact), you have to manually clone the repository somewhere, install composer dependencies (to have access to vendor/bin/dep), then run the deployment.

This feature allows to tell Deployer to automatically clone the repo and build from it (instead of deploying from the current working directory).

About why you would want to deploy from a local environment:

- If a deployment fails, it's a lot easier to debug what happens from a local env
- Depending on the context of the project, you may not necessarily want to automate your deployments with a CI

Reply:
`build_from_repo` sounds good to me